### ISSUE TITLE
pressed event is actually released event

### ISSUE THREAD

**Issue by mtempels** [bnielsen1965/rpi-gpio-buttons#7]
There is something strange going on.. 
this is my example program:

---------------8< SNIP 8<-------------
```
"use strict";

const BTN1 = 26;
const BTN2 = 20;
const gpio = require("rpi-gpio");
const buttons = require("rpi-gpio-buttons")([BTN1, BTN2]);

gpio.setMode("mode_bcm");

buttons.setTiming({
  "pressed": 50,
  "clicked": 50,
  "debounce": 25
});

buttons.on("pressed", (pin) => {
  switch (pin) {
    case BTN1:
      console.log("BIP");
      break;
    case BTN2:
      console.log("BOP");
      break;
    }
});
buttons.on("released", (pin) => {
  switch (pin) {
    case BTN1:
      console.log("BEP");
      break;
    case BTN2:
      console.log("BAP");
      break;
    }
});
```

---------------8< SNIP 8<-------------

I use the GPIO module to switch to the BCM pinning mode, but that should not change anything else. 
Now, when I press a button, nothing happens.. when I release it, the event 'BIP' or 'BOP' is shown.. This happens every time.. 

When I change the event to "released", nothing happens on the first press/release and then when I press a second time, the event is fired on button down (where i would expect it to be on button up)

when I listen to the "button_changed" event, it fires on PRESS and RELEASE

Am I doing something wrong?

Cheers

Matthijs

**Comment by bnielsen1965** [comment]
Hi @mtempels 

I think I forgot to include circuit details in the README, the button inputs should be high when the button is not pressed and pulled to ground when the button is pressed. It sounds like your button circuit may be the opposite.

Sheet 4 on the [Explorer HAT schematic](https://github.com/bnielsen1965/Explorer-HAT/raw/master/RevP3/Schematic/Explorer%20Board%20HAT%20Rev%20P3%20Schematic.pdf) is an example of a button circuit that is using a weak internal pull up but you can use an external resistor for a pull up as well.

I tested your code on a Pi Zero W with Raspbian 9.4 and node 8.11.3 and it works as expected. And you are correct, you can set the rpi-gpio mode with the rpi-gpio library instead of passing the mode in the rpi-gpio-buttons constructor and it still works.

**Comment by mtempels** [comment]
Ah, well that explains it.. I'm using the Automation Hat from Pimoroni, that has buffered inputs but they are indeed high-triggered.. So I my button press is a high input on the pin... I'm going to check if i can use another library, or perhaps you can change your library to by high- or low-triggered in the constructor? :-D 

Cheers

Matthijs

**Comment by bnielsen1965** [comment]
> perhaps you can change your library to by high- or low-triggered in the constructor?

That is a good idea. I will make that change.

**Comment by bnielsen1965** [comment]
I just published release 1.2.0 with a new option to toggle the circuit type. For your circuit use the option usePullUp: false...

```javascript
"use strict";

const BTN1 = 26;
const BTN2 = 20;
const gpio = require("rpi-gpio");
const buttons = require("rpi-gpio-buttons")([BTN1, BTN2], { usePullUp: false });

gpio.setMode("mode_bcm");

buttons.setTiming({
  "pressed": 50,
  "clicked": 50,
  "debounce": 25
});

buttons.on("pressed", (pin) => {
  switch (pin) {
    case BTN1:
      console.log("BIP");
      break;
    case BTN2:
      console.log("BOP");
      break;
    }
});
buttons.on("released", (pin) => {
  switch (pin) {
    case BTN1:
      console.log("BEP");
      break;
    case BTN2:
      console.log("BAP");
      break;
    }
});
```

**Comment by mtempels** [comment]
Ghe! Cool! Thanks for the quick fix! This is now exactly what i need :-D
And I think it is a very nice addition to the library too!
Thanks again!!

Have a great weekend! (I'm closing this 👍 )

Cheers!

Matthijs
